src/main/java/me/arminb/sara/dao/QuestionDAOImpl.java

```diff
@@ -0,0 +1,181 @@
+public class QuestionDAOImpl implements QuestionDAO {
+
+    Logger logger = LoggerFactory.getLogger(QuestionDAOImpl.class);
+
+    @Autowired
+    private MongoDatabase database;
+
+
+    private Question mapQuestion(Document questionDoc){
+
+        Question question = new Question();
+
+        DateFormat format = new SimpleDateFormat("EEE MMM d HH:mm:ss z yyyy");
+        Date createDate = null;
+        Date modifiedDate = null;
+        try {
+            if (questionDoc.get("created_date") != null){
+                createDate = format.parse(questionDoc.get("created_date").toString());
+            }
+            if (questionDoc.get("modified_date") != null) {
+                modifiedDate = format.parse(questionDoc.get("modified_date").toString());
+            }
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+
+        List<String> tagList = (List<String>) questionDoc.get("tags");
+        question.setTags(tagList);
+
+        question.setId(questionDoc.get("_id").toString());
+        question.setCreatedAt(createDate);
+        question.setModifiedAt(modifiedDate);
+        question.setContent(questionDoc.get("content").toString());
+        question.setRate(Integer.parseInt(questionDoc.get("rate").toString()));
+        question.setTitle(questionDoc.get("title").toString());
+        question.setUser(questionDoc.get("user").toString());
+
+        List<Answer> ansList = (List<Answer>) questionDoc.get("answers");
+        question.setAnswers(ansList);
+
+        return question;
+    }
+
+    @Override
+    public Question findById(String id) throws DataAccessException {
+        try {
+            MongoCollection<Document> collection = database.getCollection("questions");
+            BasicDBObject query = new BasicDBObject();
+            query.append("_id", new ObjectId(id));
+            MongoCursor<Document> cursor = collection.find(query).iterator();
+            if (cursor.hasNext()) {
+                Document questionDoc = cursor.next();
+                Question question = mapQuestion(questionDoc);
+                return question;
+            }
+            else{
+                return null;
+            }
+        } catch (MongoException e) {
+            logger.warn("Failed to fetch question {} from the database", id, e);
+            throw new DataAccessException();
+        }
+    }
+
+    @Override
+    public List<Question> find(String title, String tag, Integer pageNumber, Integer pageCount) throws DataAccessException {
+        if (pageNumber == null){
+            pageNumber = DEFAULT_PAGE_NUMBER;
+        }
+        if (pageCount == null){
+            pageCount = DEFAULT_PAGE_COUNT;
+        }
+        List<Question> list = new ArrayList();
+        try {
+            MongoCollection<Document> collection = database.getCollection("questions");
+            BasicDBObject query = new BasicDBObject();
+            if (title != null) {
+                query.append("title", title);
```

Comment:
change the query to use regex instead of equality. You can do it like this:
query.append("title", new BasicDBObject("$regex", ".*" + title + ".*"));